### ISSUE TITLE
A question

### ISSUE THREAD

**Issue by rj76** [jakobhellermann/axum_openapi#1]
Hi there, this is not an issue but a question.

The project looks unfinished, but I find it interesting because it parses axum routes and handlers in `axum_openapi_derive` but I can't seem to find where the parse macros are being used. For a test project I want to parse the axum app and see if I can generate wasm client code for the endpoints defined :)

Cheers,
Richard

**Comment by jakobhellermann** [comment]
Uh I don't really remember a lot because it's been a while since I wrote this, but I think what happend is 
- the crate started as a macro that can be attached to Axum handlers which would which would generate some code that created an openapi description for that route 
- but then after working with `bevy` I realized that I could do this without macros, by having a trait implemented by axum handler parameters (`OperationParameter`), and then implementing a trait `OperationHandler` for all functions (up to a fixed length) that would just call all the parameter's impl https://github.com/jakobhellermann/axum_openapi/blob/9458d496bdd99d9471ae207405a86c05b9899606/src/openapi_impl.rs#L24

- due to the way that the Axum `Router` is just a recursively nested generic type based on the handler functions I could implement an trait for that that collects all the handler implementations https://github.com/jakobhellermann/axum_openapi/blob/9458d496bdd99d9471ae207405a86c05b9899606/src/openapi_impl.rs#L94. I don't know if this still works in current Axum, I made this way back when Axum 0.1 got out. If this doesn't work anymore you would have to list all your handler routes manually.
- then the trait implemented on the router is called to get the actual openapi spec https://github.com/jakobhellermann/axum_openapi/blob/9458d496bdd99d9471ae207405a86c05b9899606/examples/petstore.rs#L12


TLDR: the macro isn't necessary anymore because I came up with a way that doesn't need it

**Comment by jakobhellermann** [comment]
https://github.com/jakobhellermann/axum_openapi/commit/026b65cfbc5456db0933fca4422e305694aeb90c this is the commit that implemented the trait based approach and put all the old code behind the `macro-based` feature flag

**Comment by rj76** [comment]
Thanks Jakob! I'll check it out.
